src/validation/schema.js

```diff
@@ -389,6 +389,287 @@ const validateNoImportDirective = def =>
       )
     : List()
 
+const validateNoFullTextDirective = def =>
+  def.directives.find(directive => directive.name.value == 'fulltext')
+    ? List().push(
+        immutable.fromJS({
+          loc: def.name.loc,
+          entity: def.name.value,
+          message: `@fulltext directive only allowed on '${RESERVED_TYPE}' type`,
+        }),
+      )
+    : List()
+
+const validateFullTextDirectiveFields = (def, directive) => {
+  return directive.arguments.reduce((errors, argument) => {
+    return errors.concat(
+      ['name', 'language', 'algorithm', 'include'].includes(argument.name.value)
+        ? List([])
+        : List([
+            immutable.fromJS({
+              loc: directive.name.loc,
+              entity: def.name.value,
+              message: `found invalid argument: '${argument.name.value}', @fulltext directives only allow 'name','language', 'algorithm', and 'includes' arguments`,
+            }),
+          ]),
+    )
+  }, List([]))
+}
+
+const validateFullTextDirectiveName = (def, directive) => {
+  let types = directive.arguments.find(argument => argument.name.value == 'name')
```

Comment:
Why is this called `types`?

Reply:
Good point, it'd be more accurate to refer to them as `arguments`.  

I don't think that variable needs to be created at all though since it isn't being used later in these functions; I'll just remove it.